### ISSUE TITLE
Inconsistent virtual_keycode on macOS and OpenBSD

### ISSUE THREAD

**Issue by afh** [rust-windowing/winit#2032]
While using [alacritty](https://github.com/alacritty/alacritty) on macOS and OpenBSD I notice a difference in the keybinding configuration, where some keybindings working on macOS are not working on OpenBSD.
The helpful folks in the `#alacritty` Libera.Chat channel pointed me to this repository and suggested that the issue that I run into might be related to a missing (platform) mapping in this repository, which alacritty uses as a dependency.

I modified the [`examples/resizable.rs`](https://github.com/rust-windowing/winit/blob/master/examples/resizable.rs) in order to provide a Minimal Working Example (MWE) (see [below](#mwe)).

Note that on OpenBSD I am using the `us.swapctrlcaps` keyboard layout, have assigned the physical <kbd>Control</kbd> left of the spacebar to `Hyper_L` using `keycode 37 = Hyper_L` in `~/.Xmodmap` (this should not matter though as the tests below only use the <kbd>Command</kbd> key), and have swapped the left logo key with the left alt key in order to match macOS' layout where the left <kbd>Command</kbd> key is left of the space bar and the left <kbd>Alt (Option)</kbd> key is to the left of the left <kbd>Command</kbd> key resulting in: <kbd>Alt</kbd> <kbd>Command</kbd> <kbd>Space</kbd>

|| OpenBSD | macOS |
| --- | --- | --- |
| `sw_vers -productVersion` | —  | 11.6 |
| `uname -r` | 6.9  | 20.6.0 |
| `rustc --version` | 1.51.0  | 1.51.0 |

I've tested the [MWE](#mwe) on both macOS and OpenBSD as follows:

- Place the `Cargo.toml` and `src/main.rs` files (see [below](#mwe)) in a separate directory for testing
- Build and run the MWE: `cargo build && ./target/debug/winit-key-test` on macOS and OpenBSD
- Press the keys as shown in the table below and compare the output of the MWE on macOS and OpenBSD
- I would expect the following keycodes (also highlighted in bold in the MWE Keycode Comparison table below) to be the same across platforms:

| Key(s) | OpenBSD | macOS |
| --- | --- | --- |
| <kbd>Shift</kbd> + <kbd>8</kbd> | `Asterisk` | `Key8` |
| <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>8</kbd> | `Asterisk` | `Key8` |
| <kbd>Shift</kbd> + <kbd>[</kbd> | `None` | `LBracket` |
| <kbd>Shift</kbd> + <kbd>]</kbd> | `None` | `RBracket` |
| <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>[</kbd> | `None` | `LBracket` |
| <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>]</kbd> | `None` | `RBracket` |
     
<details><summary>Detailed MWE Keycode Comparison Between OpenBSD and macOS</summary>

| Key(s) | OpenBSD | macOS |
| --- | --- | --- |
| <kbd>Command</kbd> <sub>(Key left of the spacebar)</sub>| scancode:  56, keycode: None, modifiers: LOGO | scancode:  55, keycode: Some(LWin), modifiers: LOGO  |
| <kbd>Alt</kbd> <sub>(Key left of <kbd>Command</kbd>)</sub> | scancode: 107, keycode: Some(LAlt), modifiers: ALT | scancode:  58, keycode: Some(LAlt), modifiers: ALT  |
| <kbd>Shift</kbd> | scancode:  42, keycode: Some(LShift), modifiers: SHIFT | scancode:  56, keycode: Some(LShift), modifiers: SHIFT |
| <kbd>8</kbd> | scancode:   9, keycode: Some(Key8), modifiers: (empty) | scancode:  28, keycode: Some(Key8), modifiers: (empty)  |
| <kbd>Shift</kbd> + <kbd>8</kbd> | **scancode:   9, keycode: Some(Asterisk), modifiers: SHIFT** | **scancode:  28, keycode: Some(Key8), modifiers: SHIFT** |
| <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>8</kbd> | **scancode:   9, keycode: Some(Asterisk), modifiers: SHIFT \| LOGO** | **scancode:  28, keycode: Some(Key8), modifiers: SHIFT \| LOGO** |
| <kbd>[</kbd> | scancode:  26, keycode: Some(LBracket), modifiers: (empty) | scancode:  33, keycode: Some(LBracket), modifiers: (empty)  |
| <kbd>]</kbd> | scancode:  27, keycode: Some(LBracket), modifiers: (empty) | scancode:  30, keycode: Some(LBracket), modifiers: (empty)  |
| <kbd>Shift</kbd> + <kbd>[</kbd> | **scancode:  26, keycode: None, modifiers: SHIFT** | **scancode:  33, keycode: Some(LBracket), modifiers: SHIFT** |
| <kbd>Shift</kbd> + <kbd>]</kbd> | **scancode:  26, keycode: None, modifiers: SHIFT** | **scancode:  30, keycode: Some(LBracket), modifiers: SHIFT** |
| <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>[</kbd> | **scancode:  26, keycode: None, modifiers: SHIFT \| LOGO** | **scancode:  33, keycode: Some(LBracket), modifiers: SHIFT \| LOGO** |
| <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>]</kbd> | **scancode:  27, keycode: None, modifiers: SHIFT \| LOGO** | **scancode:  30, keycode: Some(LBracket), modifiers: SHIFT \| LOGO** |

</details>

## Minimal Working Example

<details><summary><code>Cargo.toml</code></summary>

```
[package]
name = "winit-key-test"
version = "0.1.0"
edition = "2018"

[dependencies]
winit = "0.25.0"
```

</details>

<details id=mwe><summary><code>src/main.rs</code></summary>

```
// Adapted from examples/resizable.rs
use winit::{
    dpi::LogicalSize,
    event::{ElementState, Event, KeyboardInput, VirtualKeyCode, WindowEvent},
    event_loop::{ControlFlow, EventLoop},
    window::WindowBuilder,
};

fn main() {
    let event_loop = EventLoop::new();

    let mut resizable = false;

    let window = WindowBuilder::new()
        .with_title("Displaying KeyboardInput")
        .with_inner_size(LogicalSize::new(400.0, 200.0))
        .with_resizable(resizable)
        .build(&event_loop)
        .unwrap();

    event_loop.run(move |event, _, control_flow| {
        *control_flow = ControlFlow::Wait;

        match event {
            Event::WindowEvent { event, .. } => match event {
                WindowEvent::CloseRequested => *control_flow = ControlFlow::Exit,
                WindowEvent::KeyboardInput {
                    input, ..
                } => {
                    if input.state != ElementState::Pressed {
                        return;
                    }
                    println!("scancode: {scancode:>sw$}, keycode: {keycode:?}, modifiers: {modifiers:?}"
                             , scancode=input.scancode, sw=3
                             , keycode=input.virtual_keycode
                             , modifiers=input.modifiers);
                }
                _ => (),
            },
            _ => (),
        };
    });
}
```

</details>

**Comment by afh** [comment]
Going through the list of existing issues it seems that this issue should probably be included in the **Meta Issue: Keyboard input #1806** and is possibly related to **No character getting received on international keyboard on macOS #1208**.

**Comment by afh** [comment]
Thanks you so very much for addressing this! 🎉 I look forward to the next release of winit and update software (e.g. [alacritty](http://github.com/alacritty/alacritty/pulls)) using winit.
